### DISCUSSION TITLE
How to get file name when loading data from S3

### DISCUSSION THREAD

**Discussion by jvyoralek** [slingdata-io/sling-cli#270] [General]
I was using the `SLING_STREAM_URL_COLUMN` property to obtain the original filename for the row being processed from **S3**, but it appears the behavior has changed. Currently, I am receiving only the path stored in the `_sling_stream_url` column.

**Example value in the target table** (the target is a PostgreSQL database):
- before - `s3://finance/test/file1.csv`
- now -  `s3://finance/test/`

I am unsure when this change occurred, as I might have missed a few versions. The current behavior is observed with Sling version `1.2.5`.

Is there another way to store the `filename` in the **target**? It was useful in some cases for me.

**Comment by flarco** [comment]
Thanks, yes there was a process optimization done last release when
ingesting folders, but the stream name lost granularity. Should be fixed
next release.
